Task_oop_02_module_8_Students

StellaVS opened the pull request:
Составить программу для начисления стипендии студентам по результатам экзаменационной сессии. Информация о результатах сессии включает в себя:
– фамилию;
– имя;
– отчество;
– номер группы;
– экзаменационные оценки.
Количество экзаменационных оценок не менее 3 и не более 5. Стипендия начисляется студентам, сдавшим все экзамены в сессию, по следующим правилам. Студенты, сдавшие все экзамены на "отлично" получают надбавку равную 100%; студенты, сдавшие экзамены на "хорошо" и "отлично" – 50%; а студенты, сдавшие экзамены на "хорошо", – 25%. Стипендия не начисляется студентам, имеющим в сессию более двух удовлетворительных оценок.
Список студентов каждой группы, получивших стипендию, вывести на экран, упорядочив его по алфавиту.

StellaVS committed:
Create class & main

StellaVS committed:
Add system of marks

StellaVS committed:
Add methods & logic of bonus

StellaVS committed:
Delete task_9_163.java

StellaVS committed:
Delete HelloWorldJava.java

strazhkoe commented on src/dp/ort/ua/OOP_02/SystemMarks.java:
```diff
@@ -0,0 +1,21 @@
+	private static final int EXCELENT = 5;
+	private static final int GOOD = 4;
+	private static final int TERNARY = 3;
+
```
Here you can simply set constants as public and remove getters

StellaVS replied:
Getters remove, access modifier changed

`public class SystemMarks {

	public static final int EXCELENT = 5;
	public static final int GOOD = 4;
	public static final int TERNARY = 3;
`

strazhkoe commented on src/dp/ort/ua/OOP_02/Student.java:
```diff
@@ -0,0 +1,110 @@
+			System.out.println("Все оценки выставлены " + getSurname() + getPatronymic());
+	}
+
+	public int Percent(int[] Marks) {
```
please, add javadoc for this method. Also, rename it to reflect what does it do. It should start with lower-case letter. Why do we need Marks here?

StellaVS replied:
The method of filling the student assessments rewritten.
`public void setExamMarks(int[] examMarks) {
		_examMarks = new int[_sizeBook];
		if (examMarks.length <= _sizeBook) {
		for (int i = 0; i < examMarks.length; i++) {
			_examMarks[i] = examMarks[i];
						}
		} else if(examMarks.length>_sizeBook)
				System.out.println("Все оценки выставлены для студента" + getSurname() +" " + getName());
					}`

StellaVS committed:
HAPPY NEW YEAR! 
Modified Class Marks modifiers changed to public. The method of filling
the student assessments rewritten.

StellaVS committed:
restart

StellaVS committed:
*****

StellaVS committed:
Merge branch 'task_OOP_02_module_8' of https://github.com/StellaVS/Java_base_stalina.git into task_OOP_02_module_8

StellaVS committed:
restart

StellaVS committed:
The solution to the problem of commits

strazhkoe commented on src/dp/ort/ua/OOP_02/Student.java:
```diff
@@ -0,0 +1,104 @@
+		_percentStipend = percentStipend;
+	}
+
+	public int Percent(int[] Marks) {
```
#14 Please fix

strazhkoe commented on src/dp/ort/ua/OOP_02/Student_main.java:
```diff
@@ -0,0 +1,54 @@
+		}
+	}
+
+	private static String toStringStudent(Student student) {
```
#15 Please fix

StellaVS committed:
Rename method

StellaVS committed:
main rewritten

strazhkoe commented on src/dp/ort/ua/OOP_02/Student.java:
```diff
@@ -0,0 +1,117 @@
+	}
+
+	
+	public static String toStringStudent(Student student) {
```
Все хорошо. Два момента: почему тут toStringStudent а не переопределенный toString? И надо убрать из студента выводы на консоль и заменить возвращаемыми значениями

strazhkoe commented:
Все хорошо. Два момента: почему в студенте метод toStringStudent а не переопределенный toString? И надо убрать из студента выводы на консоль и заменить возвращаемыми значениями

StellaVS committed:
Override method toString, create return values
